presentation/common/src/presentation-common/content/PropertyValueFormatter.ts

```diff
@@ -62,36 +62,41 @@ export class ContentPropertyValueFormatter {
   constructor(private _koqValueFormatter: KoqPropertyValueFormatter) { }
 
   public async formatPropertyValue(field: Field, value: Value, unitSystem?: UnitSystemKey): Promise<DisplayValue> {
-    if (isFieldWithKoq(field) && typeof value === "number") {
-      const koq = field.properties[0].property.kindOfQuantity;
-      const formattedValue = await this._koqValueFormatter.format(value, { koqName: koq.name, unitSystem });
-      if (formattedValue !== undefined)
-        return formattedValue;
-    }
+    const doubleFormatter = isFieldWithKoq(field)
+      ? async (rawValue: number) => {
+        const koq = field.properties[0].property.kindOfQuantity;
+        const formattedValue = await this._koqValueFormatter.format(rawValue, { koqName: koq.name, unitSystem });
+        if (formattedValue !== undefined)
+          return formattedValue;
+        return formatDouble(rawValue);
+      }
+      : async (rawValue: number) => formatDouble(rawValue);
 
-    return this.formatValue(field.type, value);
+    return this.formatValue(field.type, value, { doubleFormatter });
   }
 
-  public formatValue(type: TypeDescription, value: Value): DisplayValue {
+  private async formatValue(type: TypeDescription, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string>}): Promise<DisplayValue> {
     switch (type.valueFormat) {
       case PropertyValueFormat.Primitive:
-        return this.formatPrimitiveValue(type, value);
+        return this.formatPrimitiveValue(type, value, ctx);
       case PropertyValueFormat.Array:
         return this.formatArrayValue(type, value);
       case PropertyValueFormat.Struct:
         return this.formatStructValue(type, value);
     }
   }
 
-  public formatPrimitiveValue(type: PrimitiveTypeDescription, value: Value) {
+  private async formatPrimitiveValue(type: PrimitiveTypeDescription, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string>}) {
     if (value === undefined)
       return "";
 
+    const formatDoubleValue = async (raw: number) => ctx ? ctx.doubleFormatter(raw) : formatDouble(raw);
+
     if (type.typeName === "point2d" && isPoint2d(value)) {
-      return `X: ${formatDouble(value.x)} Y: ${formatDouble(value.y)}`;
+      return `X: ${await formatDoubleValue(value.x)} Y: ${await formatDoubleValue(value.y)}`;
     }
     if (type.typeName === "point3d" && isPoint3d(value)) {
-      return `X: ${formatDouble(value.x)} Y: ${formatDouble(value.y)} Z: ${formatDouble(value.z)}`;
+      return `X: ${await formatDoubleValue(value.x)} Y: ${await formatDoubleValue(value.y)} Z: ${await formatDoubleValue(value.z)}`;
```

Comment:
I change the addon to separate components with a semicolon. Can you make a similar change here?